From: Andrii Nakryiko <andriin@fb.com>
Date: Mon, 25 Nov 2019 13:29:48 -0800
Subject: libbpf: Fix usage of u32 in userspace code
Patch-mainline: v5.5-rc1
Git-commit: b615e5a1e067dcb327482d1af7463268b89b1629
References: bsc#1155518

u32 is not defined for libbpf when compiled outside of kernel sources (e.g.,
in Github projection). Use __u32 instead.

Fixes: b8c54ea455dc ("libbpf: Add support to attach to fentry/fexit tracing progs")
Signed-off-by: Andrii Nakryiko <andriin@fb.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Link: https://lore.kernel.org/bpf/20191125212948.1163343-1-andriin@fb.com
Acked-by: Gary Lin <glin@suse.com>
---
 tools/lib/bpf/libbpf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/lib/bpf/libbpf.c
+++ b/tools/lib/bpf/libbpf.c
@@ -4994,7 +4994,7 @@ int libbpf_find_vmlinux_btf_id(const cha
 	char *dst = raw_tp_btf + sizeof(BTF_PREFIX) - 1;
 	const char *btf_name;
 	int err = -EINVAL;
-	u32 kind;
+	__u32 kind;
 
 	if (IS_ERR(btf)) {
 		pr_warning("vmlinux BTF is not found\n");
